### DISCUSSION TITLE
Wavesurfer Auto Gain Control

### DISCUSSION THREAD

**Discussion by gartlandsoftware** [katspaugh/wavesurfer.js#3724] [Q&A]
Hi,

When using the standard browser recording functionality (ie not using Wavesurfer) I find that Chrome automatically adjusts the level of my microphone. To resolve this, I'm able to implement the following options in my Javascript to enable/disable echoCancellation, noiseSuppression and autoGainControl:

> var constraints = {
>   audio: {
>     echoCancellation: false,
>     noiseSuppression: false,
>     autoGainControl: false,
>     volume: 1.0
>   }
> };
> navigator.mediaDevices.getUserMedia(constraints)

How can I implement these options (echoCancellation, noiseSuppression, autoGainControl) when using the audio recording functionality in Wavesurfer?

Any examples would be appreciated!

**Comment by katspaugh** [comment]
The Record plugin doesn’t yet allow passing these options to the recorder. Pull requests are welcome!

**Reply by gartlandsoftware** [reply]
Thank you... unfortunately I don't have the expertise to create a pull request for this, but I would really appreciate it if someone who does could make this happen! There is a useful list of constraints available here: https://blog.addpipe.com/audio-constraints-getusermedia/
